src/networking/backend.ts

```diff
@@ -92,12 +96,13 @@ export class Backend {
       app_user_id: appUserId,
       product_id: productId,
       email: email,
+      price_id: purchaseOption.priceId,
       presented_offering_identifier:
         presentedOfferingContext.offeringIdentifier,
     };
 
-    if (purchaseOptionId && purchaseOptionId !== "base_option") {
-      requestBody.offer_id = purchaseOptionId;
+    if (purchaseOption.id !== "base_option") {
+      requestBody.offer_id = purchaseOption.id;
```

Comment:
Not sure if we should still filter out the `base_option`... But decided to keep the same logic for now.